app/controllers/main_controller.rb

```diff
@@ -9,3 +9,45 @@
 # Your controller actions go below this line.
 # -----------------------------------------------------------------------------
 
+MyApp.get "/" do
+  # If a GET request is made to the root path, the following line of code
+  # looks for a .erb view file located in the 'views' directory at the given
+  # location. So it's going to look for views/main/welcome.erb.
+  # 
+  # Then it will combine that view file with the layout file and sent the
+  # combined document back to the client.
+  erb :"welcome"
+end
+
+
+# MyApp.get "/segment" do
+# 	  input = Word.new
+# 	  input.unsegmented = params["string_to_segment"]
+# 	  x = StringSegmenter.new(params["string_to_segment"])
+# 	  x.run_program
+# 	  input.segmented = x.final_words
+# 	  input.save
+# 	  @result = input.segmented
+#   erb :"result"
+# end
+
+MyApp.get "/segment" do
+	if  Word.where(:unsegmented => params["string_to_segment"]).blank?
+	  input = Word.new
+	  input.unsegmented = params["string_to_segment"]
+	  x = StringSegmenter.new(params["string_to_segment"])
+	  x.run_program
+	  input.segmented = x.final_words
+	  input.save
+	  @result = input.segmented
+	else
+	   @result = Word.where(:unsegmented => params["string_to_segment"]).segmented
```

Comment:
With this NoMethodError, Ruby is telling you that `segmented` cannot be called on whatever `Word.where(:unsegmented => params["string_to_segment"])` returns.

From your understanding of how ActiveRecord classes work, you know that `segmented` is a method for `Word` objects. So I assume your expectation here is that `Word.where(:unsegmented => params["string_to_segment"])` returns a `Word` object. But it would seem it is not.

So what is it?

To find out, use an interrupt:

```ruby
MyApp.get "/segment" do
  if  Word.where(:unsegmented => params["string_to_segment"]).blank?
    # ...
  else
    binding.pry
    @result = Word.where(:unsegmented => params["string_to_segment"]).segmented
    return @result
  end
  #...
end
```

When that `binding.pry` is triggered, jump to its REPL and enter in `Word.where(:unsegmented => params["string_to_segment"])`. What do you get back? A `Word` object?

See if this helps you on your way, and let me know if I can help more. Please report back if this helps.